torch/_dynamo/config.py

```diff
@@ -78,6 +78,8 @@
 # Set this to False to assume nn.Modules() contents are immutable (similar assumption as freezing)
 guard_nn_modules = False
 
+report_guard_failures = False
```

Comment:
can you also tie this into CompilationProfiler somehow?  Perhaps at least issue an error/warning when using profiler saying to enable this flag, or even better, find a way to just automatically flip it on when that backend is being used (instead of needing an actual config)